### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by zhijianli88**
They have been on my hand for a long time. Both of them are found during reviewing the code.

### CODE REVIEW

**Comment by zhijianli88** [comment]
look there was a false CI report
https://dev.azure.com/ucfconsort/f3b4a7f7-c9a7-467e-8013-95fd10fcc2ac/_apis/build/builds/33932/logs/47
where it ponits to errors when docker setup its network environment.
```
2021-12-16T09:15:13.4001192Z Docker client API version: '1.41'
2021-12-16T09:15:13.4015727Z ##[command]/usr/bin/docker ps --all --quiet --no-trunc --filter "label=d881e8"
2021-12-16T09:15:13.4499829Z ##[command]/usr/bin/docker network prune --force --filter "label=d881e8"
2021-12-16T09:15:13.4996546Z ##[command]/usr/bin/docker login --username "***" --password-stdin https://***
2021-12-16T09:15:28.7658678Z Error response from daemon: Get "https://***/v2/": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
2021-12-16T09:15:28.7813656Z ##[error]Docker login fail with exit code 1
2021-12-16T09:15:28.7826695Z ##[section]Finishing: Initialize containers
2021-12-16T09:15:28.7914743Z ##[section]Starting: Checkout
2021-12-16T09:15:28.8142321Z ==============================================================================
2021-12-16T09:15:28.8143626Z Task         : Get sources
2021-12-16T09:15:28.8144451Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2021-12-16T09:15:28.8144922Z Version      : 1.0.0
2021-12-16T09:15:28.8145446Z Author       : Microsoft
2021-12-16T09:15:28.8146199Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2021-12-16T09:15:28.8146727Z ==============================================================================
2021-12-16T09:15:29.1718367Z ##[section]Finishing: Checkout
2021-12-16T09:15:29.1739349Z ##[section]Starting: Stop Containers
2021-12-16T09:15:29.1769386Z Remove container network: vsts_network_25361034db1d47d8b3a2222d5fadaf84
2021-12-16T09:15:29.1775382Z ##[command]/usr/bin/docker network rm vsts_network_25361034db1d47d8b3a2222d5fadaf84
2021-12-16T09:15:29.2063087Z Error: No such network: vsts_network_25361034db1d47d8b3a2222d5fadaf84
2021-12-16T09:15:29.2088517Z ##[warning]Docker network rm failed with exit code 1
2021-12-16T09:15:29.2092605Z ##[section]Finishing: Stop Containers
2021-12-16T09:15:29.2140757Z ##[section]Starting: Finalize Job
2021-12-16T09:15:29.2171209Z TestResultLogParser: JasmineTestResultParser : Starting jasmine test result parser.
2021-12-16T09:15:29.2171648Z TestResultLogParser: JestTestResultParser : Starting jest test result parser.
2021-12-16T09:15:29.2172046Z TestResultLogParser: MochaTestResultParser : Starting mocha test result parser.
2021-12-16T09:15:29.2172463Z TestResultLogParser: PythonTestResultParser : Starting python test result parser.
2021-12-16T09:15:29.4712882Z TestResultLogParser: Waiting for log plugin to finish, pending process 1 log lines.
```

File: providers/rxe/rxe.c [hunk 1]
```diff
@@ -1528,7 +1528,7 @@ static int post_one_send(struct rxe_qp *qp, struct rxe_wq *sq,
 
 	err = validate_send_wr(qp, ibwr, length);
 	if (err) {
-		printf("validate send failed\n");
+		fprintf(stderr, "rxe: validate send failed\n");
```
Comment (rleon) [inline providers/rxe/rxe.c:1531]: These prints or should be deleted to converted to general log infrastructure.

See efa as an example.

File: libibverbs/dynamic_driver.c [hunk 2]
```diff
@@ -230,6 +231,7 @@ void load_drivers(void)
 			while ((env_name = strsep(&list, ":;")))
 				load_driver(env_name);
 		}
+		free(list);
```
Comment (rleon) [inline libibverbs/dynamic_driver.c:234]: There are two calls to strdupa() here and all of them allocate memory, so if you want to be super-pedantic, you will need to call to free after every strdupa.

**Comment by rleon** [comment]
No empty commit messages please.

File: libibverbs/dynamic_driver.c (hunk 2 above)
Comment (zhijianli88) [reply libibverbs/dynamic_driver.c:234]: > There are two calls to strdupa() here and all of them allocate memory, so if you want to be super-pedantic, you will need to call to free after every strdupa.

I guess you mean below line226 and line 230. IIUC, these 2 strdupa() are mutually exclusive.

```
222         if (getuid() == geteuid()) {                                            
223                 char *list = NULL, *env_name;                                   
224                                                                                 
225                 if ((env = getenv("RDMAV_DRIVERS"))) {                          
226                         list = strdupa(env);                                    
227                         while ((env_name = strsep(&list, ":;")))                
228                                 load_driver(env_name);                          
229                 } else if ((env = getenv("IBV_DRIVERS"))) {                     
230                         list = strdupa(env);                                    
231                         while ((env_name = strsep(&list, ":;")))                
232                                 load_driver(env_name);                          
233                 }                                                               
234                 free(list);                                                     
235         }   
```

BTW, i think we could simplify the code a abit where there are some duplicated logic.

**Comment by zhijianli88** [comment]
Append one more patch to this PR, which will simplify the code first.

File: libibverbs/dynamic_driver.c (hunk 2 above)
Comment (zhijianli88) [reply libibverbs/dynamic_driver.c]: > you will need to call to free after every strdupa.
My bad, i missleading *strdupa* to *strdup*, where strdupa(3) are automatically free

currently, i only keep one patch in this PR.

**Pull request merged**